compiler/rustc_error_messages/src/lib.rs

```diff
@@ -142,7 +142,7 @@ pub fn fluent_bundle(
     // If the user requests the default locale then don't try to load anything.
     if let Some(requested_locale) = requested_locale {
         let mut found_resources = false;
-        for sysroot in user_provided_sysroot.iter_mut().chain(sysroot_candidates.iter_mut()) {
+        for mut sysroot in Some(user_provided_sysroot).into_iter().chain(sysroot_candidates.into_iter()) {
```

Comment:
I think the default "materialized" sysroot will appear in this chain twice now, but that's fine.

Reply:
Indeed. I'm currently looking into getting rid of the need for sysroot fallbacks entirely.